src/Type/ClosureType.php

```diff
@@ -97,6 +97,11 @@ public function getReferencedClasses(): array
 		return array_merge($classes, $this->returnType->getReferencedClasses());
 	}
 
+	public function getObjectClassNames(): array
```

Comment:
another maybe dumb question:

before this PR the `ClosureType` was not considered a `TypeWithClassName`, additionally it was not handled in `TypeUtil::getDirectClassNames()` etc.

but we are now using `Type->getObjectClassNames()` in a few places where we do not handle/did not handle the `ClosureType` type before, but it is now returned.

does this mean we have new features/new edge cases in this area which require new tests?